rv/addons/lib/lib_carma2_json.sqf

```diff
@@ -0,0 +1,295 @@
+                };
+                default {}; // Do nothing
+            };
+            _value = _value + toString [_x];
+        }
+    } foreach _characters;
+    if (_value != "") then {
+        _array pushback([_value] call CBA_fnc_trim);
+    };
+    {
+        if (typeName _x == "STRING") then {
+            if (_x select [0,1] == "{") then {
+                _array set[_forEachIndex, JSON.parse(_x)];
+            } else {
+                if (_x select [0,1] == "[") then {
+                    _array set[_forEachIndex, JSON.parseArray(_x)];
+                } else {
+                    if (_x select [0, 1] == '"') then {
+                        private _val = toArray _x;
+                        _val deleteAt 0;
+                        _val deleteAt (count _val -1);
+                        _array set[_forEachIndex, toString _val];
+                    } else {
+                        if (_x in ["true", "false"]) then {
+                            _array set[_forEachIndex, _x == "true"];
+                        } else {
+                            _array set[_forEachIndex, parseNumber _x];
+                        };
+                    };
+                };
+            };
+        };
+    } forEach _array;
+
+    _array;
+};
+
+JSON.validate = {
+    params ["_object"];
+    true;
+};
+
+// Convert carma object to JSON string
+JSON.stringify = {
+    params ["_object"];
+
+    // Collect all variables we want to set
+    private _allVars = (allVariables _object) - ["__init", "__id", "__handles", "__prototype", "_haskey", "_keys", "_delete"];
+    private _count = (count _allVars) - 1;
+
+    // Construct JSON string
+    private _stringified = "{";
+    {
+        private _val = _thisObj.toJsonStr(_object{_x});
+        if (!isNil "_val" && {_val != ''}) then {
+            _stringified = _stringified + (format['"%1":%2', _x, _val]);
+            if (_forEachIndex < _count) then {
+                _stringified = _stringified + ",";
+            };
+        };
+    } forEach _allVars;
+    _stringified + "}";
+};
+
+JSON.toJsonStr = {
+    params["_value"];
+    if (isNil "_value") exitwith {''};
+
+    switch (typeName _value) do {
+        case ("ARRAY"): {
+            private _ret = "[";
+            private _count = count _value - 1;
+            {
+                private _element = (_thisObj.toJsonStr(_x));
+                if (!isNil "_element" && {_element != ''}) then {
+                    _ret = _ret + _element;
+                    if (_forEachIndex < _count) then {
+                        _ret = _ret + ",";
+                    };
+                };
+            } forEach _value;
+            _ret + "]";
+        };
+        case ("CODE"): {''}; // exclude code and functions from json string
+        case ("OBJECT"): {
+            private _object = new JSONObject();
+            private _objInfo = new JSONObject();
+
+            {
+                _object setvariable [_x, _value getvariable _x];
```

Comment:
You can get/set dynamic members on objects using the `{}` operator. For example this line would be: `_object{_x} =  _value{_x};`